rules/properImportRequireRule.ts

```diff
@@ -32,6 +32,10 @@ function loadMainFile(module_name: string): string | undefined {
     if (!fs.existsSync(main_file_path)) return;
   }
 
+  if (fs.lstatSync(main_file_path).isDirectory()) {
+    main_file_path = path.join(main_file_path, 'index.js');
```

Comment:
Need to get `package.json`'s `main` key, otherwise default to `index.js`... I wonder if there's any method for doing that already with TypeScript or TSLint?

Reply:
Also if you go the path of re implementing that yourself (you should really find something that does it already), if `main` is a directory then you need to add `index.js` to it again